public/app/features/dashboard-scene/saving/useSaveDashboard.ts

```diff
@@ -20,8 +20,26 @@ export function useSaveDashboard(isCopy = false) {
   const [saveDashboardRtkQuery] = useSaveDashboardMutation();
 
   const [state, onSaveDashboard] = useAsyncFn(
-    async (scene: DashboardScene, saveModel: Dashboard, options: SaveDashboardOptions) => {
+    async (
+      scene: DashboardScene,
+      options: SaveDashboardOptions &
+        SaveDashboardAsOptions & {
+          // When provided, will take precedence over the scene's save model
+          rawDashboardJSON?: Dashboard;
+        }
+    ) => {
```

Comment:
Second part of the fundamental changes: this functon now accepts only DashboardScene